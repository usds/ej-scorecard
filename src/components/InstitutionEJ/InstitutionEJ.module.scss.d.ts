declare namespace InstitutionEJNamespace {
  export interface IInstitutionEJScss {
    h2: string;
  }
}

declare const InstitutionEJScssModule: InstitutionEJNamespace.IInstitutionEJScss & {
  /** WARNING: Only available when "css-loader" is used without "style-loader" or "mini-css-extract-plugin" */
  locals: InstitutionEJNamespace.IInstitutionEJScss;
};

export = InstitutionEJScssModule;
