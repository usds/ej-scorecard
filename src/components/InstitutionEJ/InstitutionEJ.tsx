import React from 'react';
import {
  Grid,
  SummaryBox,
  SummaryBoxContent,
  SummaryBoxHeading,
  Table,
} from '@trussworks/react-uswds';

// import * as styles from './InstitutionEJ.module.scss';

import { InstitutEjProps } from '@/types';

const InstitutionEJ: React.FC<InstitutEjProps> = ({ institutEj }) => {
  const table1Data = (
    <>
      <tbody>
        <tr>
          <td colSpan={5} className="font-mono-sm text-tabular text-left">
            {institutEj.node.Row11}
          </td>
          <td className="font-mono-sm text-tabular text-left">
            {institutEj.node.Row12}
          </td>
          <td className="font-mono-sm text-tabular text-left">
            {institutEj.node.Row13}
          </td>
        </tr>
        <tr>
          <td colSpan={5} className="font-mono-sm text-tabular text-left">
            {institutEj.node.Row21}
          </td>
          <td className="font-mono-sm text-tabular text-left">
            {institutEj.node.Row22}
          </td>
          <td className="font-mono-sm text-tabular text-left">
            {institutEj.node.Row23}
          </td>
        </tr>
        <tr>
          <td colSpan={5} className="font-mono-sm text-tabular text-left">
            {institutEj.node.Row31}
          </td>
          <td colSpan={2} className="font-mono-sm text-tabular text-left">
            {institutEj.node.Row32}
          </td>
        </tr>
        <tr>
          <td colSpan={5} className="font-mono-sm text-tabular text-left">
            {institutEj.node.Row41}
          </td>
          <td colSpan={2} className="font-mono-sm text-tabular text-left">
            {institutEj.node.Row42}
          </td>
        </tr>
        <tr>
          <td colSpan={5} className="font-mono-sm text-tabular text-left">
            {institutEj.node.Row51}
          </td>
          <td colSpan={2} className="font-mono-sm text-tabular text-left">
            {institutEj.node.Row52}
          </td>
        </tr>
        <tr>
          <td colSpan={5} className="font-mono-sm text-tabular text-left">
            {institutEj.node.Row61}
          </td>
          <td colSpan={2} className="font-mono-sm text-tabular text-left">
            {institutEj.node.Row62}
          </td>
        </tr>
        <tr>
          <td colSpan={5} className="font-mono-sm text-tabular text-left">
            {institutEj.node.Row71}
          </td>
          <td colSpan={2} className="font-mono-sm text-tabular text-left">
            {institutEj.node.Row72}
          </td>
        </tr>
        <tr>
          <td colSpan={5} className="font-mono-sm text-tabular text-left">
            {institutEj.node.Row81}
          </td>
          <td colSpan={2} className="font-mono-sm text-tabular text-left">
            {institutEj.node.Row82}
          </td>
        </tr>
        <tr>
          <td colSpan={5} className="font-mono-sm text-tabular text-left">
            {institutEj.node.Row91}
          </td>
          <td colSpan={2} className="font-mono-sm text-tabular text-left">
            {institutEj.node.Row92}
          </td>
        </tr>
        <tr>
          <td colSpan={5} className="font-mono-sm text-tabular text-left">
            {institutEj.node.Row101}
          </td>
          <td colSpan={2} className="font-mono-sm text-tabular text-left">
            {institutEj.node.Row102}
          </td>
        </tr>
        <tr>
          <td colSpan={5} className="font-mono-sm text-tabular text-left">
            {institutEj.node.Row111}
          </td>
          <td colSpan={2} className="font-mono-sm text-tabular text-left">
            {institutEj.node.Row112}
          </td>
        </tr>
        <tr>
          <td colSpan={5} className="font-mono-sm text-tabular text-left">
            {institutEj.node.Row121}
          </td>
          <td colSpan={2} className="font-mono-sm text-tabular text-left">
            {institutEj.node.Row122}
          </td>
        </tr>
        <tr>
          <td colSpan={5} className="font-mono-sm text-tabular text-left">
            {institutEj.node.Row131}
          </td>
          <td colSpan={2} className="font-mono-sm text-tabular text-left">
            {institutEj.node.Row132}
          </td>
        </tr>
        <tr>
          <td colSpan={5} className="font-mono-sm text-tabular text-left">
            {institutEj.node.Row141}
          </td>
          <td colSpan={2} className="font-mono-sm text-tabular text-left">
            {institutEj.node.Row142}
          </td>
        </tr>
        <tr>
          <td colSpan={5} className="font-mono-sm text-tabular text-left">
            {institutEj.node.Row151}
          </td>
          <td colSpan={2} className="font-mono-sm text-tabular text-left">
            {institutEj.node.Row152}
          </td>
        </tr>
      </tbody>
    </>
  );

  const table2Data = (
    <>
      <tbody>
        <tr>
          <td colSpan={10} className="font-mono-sm text-tabular text-left">
            {institutEj.node.Row161}
          </td>
          <td className="font-mono-sm text-tabular text-left">
            {institutEj.node.Row162}
          </td>
        </tr>
      </tbody>
    </>
  );

  const table3Data = (
    <>
      <tbody>
        <tr>
          <td className="font-mono-sm text-tabular text-left">
            {institutEj.node.Row171}
          </td>
          <td className="font-mono-sm text-tabular text-left">
            {institutEj.node.Row172}
          </td>
        </tr>
      </tbody>
    </>
  );
  return (
    <>
      <Grid row>
        <Grid desktop={{ col: 8 }} tablet={{ col: 10 }} col={12}>
          <h2 className={`scorecard-header2`}>
            Institutionalizing Environmental Justice
          </h2>
        </Grid>
      </Grid>
      <Grid row>
        <Grid desktop={{ col: 6 }} tablet={{ col: 10 }} col={12}>
          Short narrative about the why Institutionalizing Environmental Justice
          is important. I prem ins _dolor sit amet concertetur adinissing
          plitvam at magna vitan prat convall bandit. Phasellus id urna
          ultrices, vestibulum guam a. accumsan mauris
        </Grid>
      </Grid>
      <Grid row gap={6}>
        <Grid desktop={{ col: 8 }} tablet={{ col: 10 }} col={12}>
          <Table fullWidth bordered striped>
            {table1Data}
          </Table>
          <Table fullWidth bordered striped>
            {table2Data}
          </Table>
          <Table fullWidth bordered striped>
            {table3Data}
          </Table>
        </Grid>
        <Grid desktop={{ col: 4 }} tablet={{ col: 10 }} col={12}>
          <SummaryBox className={`margin-top-2`}>
            <SummaryBoxHeading headingLevel="h3">
              {`Institutionalizing EJ at ${institutEj.node.Agency_Name}`}
            </SummaryBoxHeading>
            <SummaryBoxContent>{institutEj.node.Summary}</SummaryBoxContent>
          </SummaryBox>
        </Grid>
      </Grid>
    </>
  );
};

export default InstitutionEJ;
