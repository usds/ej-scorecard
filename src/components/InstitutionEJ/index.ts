import InstitutionEJ from './InstitutionEJ';
export default InstitutionEJ;
