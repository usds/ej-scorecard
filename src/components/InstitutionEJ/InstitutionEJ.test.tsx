import createComponentWithIntl from '@/test/testHelpers';

import InstitutionEJ from './InstitutionEJ';

const institutEj = {
  node: {
    id: `1b11ec85-da83-5b4c-a248-2c6438dcf3fe`,
    Agency_Name: `T2 Agency Dept Office`,
    Summary: `Lorem T2 ipsum dolor sit amet, consectetur adipiscing elit. Morbi luctus velit non justo tempus, sit amet bibendum tellus convallis. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Nam in gravida mi. Nam nisl ante, ultricies et massa id, euismod sodales arcu. Sed velit felis, tristique a nunc sit amet, rutrum iaculis lectus. Nullam auctor justo at odio sagittis rutrum. Nunc eget malesuada mauris. Donec tincidunt ex in mi ultrices sodales. Aliquam at lobortis eros, eu mollis tellus. Nullam sit amet elementum tellus. Suspendisse viverra ligula neque, id dapibus nisl elementum et. Integer mattis magna ac lorem fermentum, laoreet accumsan tellus tincidunt. Donec bibendum viverra velit vel eleifend.`,
    Row11: `Y-N Questn T2`,
    Row12: `Boolean`,
    Row13: `#`,
    Row21: `Y-N Questn T2`,
    Row22: `Boolean`,
    Row23: `#`,
    Row31: `SubmetricT2`,
    Row32: `#`,
    Row41: `SubmetricT2`,
    Row42: `#`,
    Row51: `SubmetricT2`,
    Row52: `#`,
    Row61: `SubmetricT2`,
    Row62: `#`,
    Row71: `Y-N Questn T2`,
    Row72: `Boolean`,
    Row81: `Y-N Questn T2`,
    Row82: `#`,
    Row91: `Y-N Questn T2`,
    Row92: `Boolean`,
    Row101: `Y-N Questn T2`,
    Row102: `#`,
    Row111: `Y-N Questn T2`,
    Row112: `Boolean`,
    Row121: `Y-N Questn T2`,
    Row122: `Boolean`,
    Row131: `Y-N Questn T2`,
    Row132: `Boolean`,
    Row141: `Y-N Questn T2`,
    Row142: `Boolean`,
    Row151: `Y-N Questn T2`,
    Row152: `#`,
    Row161: `SummaryT2`,
    Row162: `ipsum dolor sit amet, consectetur adipiscing elit. Morbi luctus velit non justo tempus, sit amet bibendum tellus convallis.`,
    Row171: `SummaryT2`,
    Row172: `ipsum dolor sit amet, consectetur adipiscing elit. Morbi luctus velit non justo tempus, sit amet bibendum tellus convallis.`,
  },
};

describe(`rendering of InstitutionEJ Component`, () => {
  it(`checks if component renders`, () => {
    const component = createComponentWithIntl(
      <InstitutionEJ institutEj={institutEj} />,
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
