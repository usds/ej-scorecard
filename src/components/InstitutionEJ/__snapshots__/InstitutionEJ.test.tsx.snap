// Vitest Snapshot v1

exports[`rendering of InstitutionEJ Component > checks if component renders 1`] = `
[
  <div
    className="grid-row"
    data-testid="grid"
  >
    <div
      className="grid-col-12 tablet:grid-col-10 desktop:grid-col-8"
      data-testid="grid"
    >
      <h2
        className="scorecard-header2"
      >
        Institutionalizing Environmental Justice
      </h2>
    </div>
  </div>,
  <div
    className="grid-row"
    data-testid="grid"
  >
    <div
      className="grid-col-12 tablet:grid-col-10 desktop:grid-col-6"
      data-testid="grid"
    >
      Short narrative about the why Institutionalizing Environmental Justice is important. I prem ins _dolor sit amet concertetur adinissing plitvam at magna vitan prat convall bandit. Phasellus id urna ultrices, vestibulum guam a. accumsan mauris
    </div>
  </div>,
  <div
    className="grid-row grid-gap-6"
    data-testid="grid"
  >
    <div
      className="grid-col-12 tablet:grid-col-10 desktop:grid-col-8"
      data-testid="grid"
    >
      <table
        className="usa-table src-components-Table-Table-module__fullwidth--fOyj6 usa-table--striped"
        data-testid="table"
      >
        <tbody>
          <tr>
            <td
              className="font-mono-sm text-tabular text-left"
              colSpan={5}
            >
              Y-N Questn T2
            </td>
            <td
              className="font-mono-sm text-tabular text-left"
            >
              Boolean
            </td>
            <td
              className="font-mono-sm text-tabular text-left"
            >
              #
            </td>
          </tr>
          <tr>
            <td
              className="font-mono-sm text-tabular text-left"
              colSpan={5}
            >
              Y-N Questn T2
            </td>
            <td
              className="font-mono-sm text-tabular text-left"
            >
              Boolean
            </td>
            <td
              className="font-mono-sm text-tabular text-left"
            >
              #
            </td>
          </tr>
          <tr>
            <td
              className="font-mono-sm text-tabular submetric"
              colSpan={5}
            >
              SubmetricT2
            </td>
            <td
              className="font-mono-sm text-tabular text-left"
              colSpan={2}
            >
              #
            </td>
          </tr>
          <tr>
            <td
              className="font-mono-sm text-tabular submetric"
              colSpan={5}
            >
              SubmetricT2
            </td>
            <td
              className="font-mono-sm text-tabular text-left"
              colSpan={2}
            >
              #
            </td>
          </tr>
          <tr>
            <td
              className="font-mono-sm text-tabular submetric"
              colSpan={5}
            >
              SubmetricT2
            </td>
            <td
              className="font-mono-sm text-tabular text-left"
              colSpan={2}
            >
              #
            </td>
          </tr>
          <tr>
            <td
              className="font-mono-sm text-tabular submetric"
              colSpan={5}
            >
              SubmetricT2
            </td>
            <td
              className="font-mono-sm text-tabular text-left"
              colSpan={2}
            >
              #
            </td>
          </tr>
          <tr>
            <td
              className="font-mono-sm text-tabular text-left"
              colSpan={5}
            >
              Y-N Questn T2
            </td>
            <td
              className="font-mono-sm text-tabular text-left"
              colSpan={2}
            >
              Boolean
            </td>
          </tr>
          <tr>
            <td
              className="font-mono-sm text-tabular text-left"
              colSpan={5}
            >
              Y-N Questn T2
            </td>
            <td
              className="font-mono-sm text-tabular text-left"
              colSpan={2}
            >
              #
            </td>
          </tr>
          <tr>
            <td
              className="font-mono-sm text-tabular text-left"
              colSpan={5}
            >
              Y-N Questn T2
            </td>
            <td
              className="font-mono-sm text-tabular text-left"
              colSpan={2}
            >
              Boolean
            </td>
          </tr>
          <tr>
            <td
              className="font-mono-sm text-tabular text-left"
              colSpan={5}
            >
              Y-N Questn T2
            </td>
            <td
              className="font-mono-sm text-tabular text-left"
              colSpan={2}
            >
              #
            </td>
          </tr>
          <tr>
            <td
              className="font-mono-sm text-tabular text-left"
              colSpan={5}
            >
              Y-N Questn T2
            </td>
            <td
              className="font-mono-sm text-tabular text-left"
              colSpan={2}
            >
              Boolean
            </td>
          </tr>
          <tr>
            <td
              className="font-mono-sm text-tabular text-left"
              colSpan={5}
            >
              Y-N Questn T2
            </td>
            <td
              className="font-mono-sm text-tabular text-left"
              colSpan={2}
            >
              Boolean
            </td>
          </tr>
          <tr>
            <td
              className="font-mono-sm text-tabular text-left"
              colSpan={5}
            >
              Y-N Questn T2
            </td>
            <td
              className="font-mono-sm text-tabular text-left"
              colSpan={2}
            >
              Boolean
            </td>
          </tr>
          <tr>
            <td
              className="font-mono-sm text-tabular text-left"
              colSpan={5}
            >
              Y-N Questn T2
            </td>
            <td
              className="font-mono-sm text-tabular text-left"
              colSpan={2}
            >
              Boolean
            </td>
          </tr>
          <tr>
            <td
              className="font-mono-sm text-tabular text-left"
              colSpan={5}
            >
              Y-N Questn T2
            </td>
            <td
              className="font-mono-sm text-tabular text-left"
              colSpan={2}
            >
              #
            </td>
          </tr>
        </tbody>
      </table>
      <table
        className="usa-table src-components-Table-Table-module__fullwidth--fOyj6 usa-table--striped"
        data-testid="table"
      >
        <tbody>
          <tr>
            <td
              className="font-mono-sm text-tabular text-left"
              colSpan={10}
            >
              SummaryT2
            </td>
            <td
              className="font-mono-sm text-tabular text-left"
            >
              ipsum dolor sit amet, consectetur adipiscing elit. Morbi luctus velit non justo tempus, sit amet bibendum tellus convallis.
            </td>
          </tr>
          <tr>
            <td
              className="font-mono-sm text-tabular text-left"
              colSpan={10}
            >
              SummaryT2
            </td>
            <td
              className="font-mono-sm text-tabular text-left"
            >
              ipsum dolor sit amet, consectetur adipiscing elit. Morbi luctus velit non justo tempus, sit amet bibendum tellus convallis.
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      className="grid-col-12 tablet:grid-col-10 desktop:grid-col-4"
      data-testid="grid"
    >
      <div
        className="usa-summary-box margin-top-2"
        data-testid="summary-box"
      >
        <div
          className="usa-summary-box__body"
        >
          <h3
            className="usa-summary-box__heading"
          >
            Institutionalizing EJ at T2 Agency Dept Office
          </h3>
          <div
            className="usa-summary-box__text"
          >
            Lorem T2 ipsum dolor sit amet, consectetur adipiscing elit. Morbi luctus velit non justo tempus, sit amet bibendum tellus convallis. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Nam in gravida mi. Nam nisl ante, ultricies et massa id, euismod sodales arcu. Sed velit felis, tristique a nunc sit amet, rutrum iaculis lectus. Nullam auctor justo at odio sagittis rutrum. Nunc eget malesuada mauris. Donec tincidunt ex in mi ultrices sodales. Aliquam at lobortis eros, eu mollis tellus. Nullam sit amet elementum tellus. Suspendisse viverra ligula neque, id dapibus nisl elementum et. Integer mattis magna ac lorem fermentum, laoreet accumsan tellus tincidunt. Donec bibendum viverra velit vel eleifend.
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
